utilities/backupable/backupable_db.cc

```diff
@@ -1373,6 +1373,7 @@ Status BackupEngineImpl::CopyOrCreateFile(
     uint64_t size_limit, std::function<void()> progress_callback,
     std::string* db_id, std::string* db_session_id) {
   assert(src.empty() != contents.empty());
+  assert(db_id != nullptr || db_session_id != nullptr);
```

Comment:
Sometimes we have explicit behaviors for release builds for cases in which an assertion would fail in a debug build, but I don't think that's needed here.

(a) This assertions says "a precondition of this function is that at least one is not null".

(b) The declaration of this function has default parameters with both null, so says "it's ok to pass both null."

(c) The new 'if' condition below implies the condition needs to be checked, so must not be a precondition of calling this function.

Which way do you want to fix so that everything agrees?

Reply:
I'd choose (b). When null is passed in for db_id  or db_session in CopyOrCreateFile, it indicates the caller does not want to obtain the value of db_id or db_session_id. 

GetFileDbIdentities can be called whenever db_id or db_session_id is not null.

I will the remove the assertions. Before calling GetFileDbIdentities in CopyOrCreateFile, the condition that checks if one of the two ids is not null will be kept.

What do you think?

Reply:
I meant (a), (b), and (c) as pieces of evidence, not options for fixing. ;)

I support removing this assertion but keeping the one in GetFileDbIdentities, because that function doesn't make sense if both are null.